### PR TITLE
Upload yaml during CI release job

### LINKED ISSUE #898: Automate yaml upload for release
For alpha we uploaded yaml manually. It should be part of release job on CI.

### PR DESCRIPTION
**Pull request by artemnikitin**
Uploads yaml as part of CI release job
close #898

### CODE REVIEW

**Comment by artemnikitin** [comment]
> Were you able to test that it works?

Locally yes.

File: build/ci/Makefile [hunk 1]
```diff
@@ -77,6 +87,13 @@ ci-release: vault-public-key vault-docker-creds
     	cloud-on-k8s-ci-release \
     	bash -c "make -C operators ci-release"
 
+yaml-upload: vault-aws-creds
+	@ AWS_ACCESS_KEY_ID=$(shell cat $(VAULT_AWS_ACCESS_KEY_FILE)) \
+		AWS_SECRET_ACCESS_KEY=$(shell cat $(VAULT_AWS_SECRET_KEY_FILE)) \
+		aws s3 cp \
+		$(ROOT_DIR)/operators/config/all-in-one.yaml \
+		s3://download.elasticsearch.org/downloads/eck/$(TAG_NAME)/all-in-one.yaml
```
Comment (pebrc) [inline build/ci/Makefile]: What about the other deployment manifests (single namespace, mixed)? Can be done in another PR I guess but maybe raise another issue if it is not happening in this one.

File: build/ci/Makefile [hunk 2]
```diff
@@ -43,6 +46,13 @@ vault-docker-creds:
 	@ VAULT_TOKEN=$$(vault write -field=token auth/approle/login role_id=$(VAULT_ROLE_ID) secret_id=$(VAULT_SECRET_ID)) \
 		vault read -address=$(VAULT_ADDR) -field=password $(VAULT_DOCKER_CREDENTIALS) > $(DOCKER_CREDENTIALS_FILE)
 
+# reads AWS creds for yaml upload
+vault-aws-creds:
+	@ VAULT_TOKEN=$$(vault write -field=token auth/approle/login role_id=$(VAULT_ROLE_ID) secret_id=$(VAULT_SECRET_ID)) \
+    	vault read -address=$(VAULT_ADDR) -field=access-key-id $(VAULT_AWS_CREDS) > $(VAULT_AWS_ACCESS_KEY_FILE)
+	@ VAULT_TOKEN=$$(vault write -field=token auth/approle/login role_id=$(VAULT_ROLE_ID) secret_id=$(VAULT_SECRET_ID)) \
```
[the remark this answers is not in the document]
Comment (artemnikitin) [reply build/ci/Makefile]: I implemented token generation once per Makefile

**Comment by artemnikitin** [comment]
jenkins test this please

**Comment by artemnikitin** [comment]
jenkins test this please

File: build/ci/Makefile (hunk 1 above)
Comment (artemnikitin) [reply build/ci/Makefile]: Created an issue for that https://github.com/elastic/cloud-on-k8s/issues/912

**Pull request merged**